cwms-data-api/src/main/java/cwms/cda/security/OpenIDAccessManager.java

```diff
@@ -69,15 +72,26 @@ public void manage(Handler handler, @NotNull Context ctx, @NotNull Set<RouteRole
     private DataApiPrincipal getUserFromToken(Context ctx) throws CwmsAuthException {
         try {
             Jws<Claims> token = jwtParser.parseClaimsJws(getToken(ctx));
-            String username = token.getBody().get("preferred_username",String.class);
+            Claims claims = token.getBody();
+            final String issuer = claims.getIssuer();
+            final String subject = claims.getSubject();
+            final String username = issuer + "::" + subject;
             AuthDao dao = AuthDao.getInstance(JooqDao.getDslContext(ctx),ctx.attribute(ApiServlet.OFFICE_ID));
-            String edipiStr = username.substring(username.lastIndexOf(".") + 1);
-            long edipi = Long.parseLong(edipiStr);
-            return dao.getPrincipalFromEdipi(edipi);
+            Optional<DataApiPrincipal> principal = dao.getPrincipalFromPrincipal(username);
+            if (principal.isPresent()) {
+                return principal.get();
+            } else if (createUsers) {
+                final String preferredUserName = claims.get("preferred_username", String.class);
+                final String givenName = claims.get("given_name", String.class);
+                final String email = claims.get("email", String.class);
+                return dao.createUser(preferredUserName,username,givenName, email);
+            } else {
+                throw new CwmsAuthException("Not Authorized",HttpServletResponse.SC_UNAUTHORIZED);
+            }
```

Comment:
some spaces after commas here would make this look better.   Those string literals should get moved into constants